packages/sdk38/src/lcdclient.ts

```diff
@@ -0,0 +1,350 @@
+    setupModuleC: LcdModuleSetup<C>,
+    setupModuleD: LcdModuleSetup<D>,
+    setupModuleE: LcdModuleSetup<E>,
+    setupModuleF: LcdModuleSetup<F>,
+  ): LcdClient & A & B & C & D & E & F;
+
+  /** Constructs an LCD client with 7 modules */
+  public static withModules<
+    A extends LcdModule,
+    B extends LcdModule,
+    C extends LcdModule,
+    D extends LcdModule,
+    E extends LcdModule,
+    F extends LcdModule,
+    G extends LcdModule
+  >(
+    options: LcdClientBaseOptions,
+    setupModuleA: LcdModuleSetup<A>,
+    setupModuleB: LcdModuleSetup<B>,
+    setupModuleC: LcdModuleSetup<C>,
+    setupModuleD: LcdModuleSetup<D>,
+    setupModuleE: LcdModuleSetup<E>,
+    setupModuleF: LcdModuleSetup<F>,
+    setupModuleG: LcdModuleSetup<G>,
+  ): LcdClient & A & B & C & D & E & F & G;
+
+  /** Constructs an LCD client with 8 modules */
+  public static withModules<
+    A extends LcdModule,
+    B extends LcdModule,
+    C extends LcdModule,
+    D extends LcdModule,
+    E extends LcdModule,
+    F extends LcdModule,
+    G extends LcdModule,
+    H extends LcdModule
+  >(
+    options: LcdClientBaseOptions,
+    setupModuleA: LcdModuleSetup<A>,
+    setupModuleB: LcdModuleSetup<B>,
+    setupModuleC: LcdModuleSetup<C>,
+    setupModuleD: LcdModuleSetup<D>,
+    setupModuleE: LcdModuleSetup<E>,
+    setupModuleF: LcdModuleSetup<F>,
+    setupModuleG: LcdModuleSetup<G>,
+    setupModuleH: LcdModuleSetup<H>,
+  ): LcdClient & A & B & C & D & E & F & G & H;
+
+  public static withModules<
+    A extends LcdModule,
+    B extends LcdModule,
+    C extends LcdModule,
+    D extends LcdModule,
+    E extends LcdModule,
+    F extends LcdModule,
+    G extends LcdModule,
+    H extends LcdModule
+  >(
+    options: LcdClientBaseOptions,
+    setupModuleA?: LcdModuleSetup<A>,
+    setupModuleB?: LcdModuleSetup<B>,
+    setupModuleC?: LcdModuleSetup<C>,
+    setupModuleD?: LcdModuleSetup<D>,
+    setupModuleE?: LcdModuleSetup<E>,
+    setupModuleF?: LcdModuleSetup<F>,
+    setupModuleG?: LcdModuleSetup<G>,
+    setupModuleH?: LcdModuleSetup<H>,
+  ): any {
+    const client = new LcdClient(options.apiUrl, options.broadcastMode);
+
+    const modules = new Array<LcdModule>();
+    if (setupModuleA) modules.push(setupModuleA(client));
+    if (setupModuleB) modules.push(setupModuleB(client));
+    if (setupModuleC) modules.push(setupModuleC(client));
+    if (setupModuleD) modules.push(setupModuleD(client));
+    if (setupModuleE) modules.push(setupModuleE(client));
+    if (setupModuleF) modules.push(setupModuleF(client));
+    if (setupModuleG) modules.push(setupModuleG(client));
+    if (setupModuleH) modules.push(setupModuleH(client));
+    for (const module of modules) {
+      assert(isNonNullObject(module), `Module must be a non-null object`);
+      for (const key in module) {
+        assert(typeof key == "string", `Found non-string module key: ${key}`);
+        (client as any)[key] = module[key];
```

Comment:
Ah, nice for easy lookup.

Is there an example of how to use this?

`(client as any).wasmd.listCodes()`

Reply:
Oh, wait, you basically copy all the functions to the top level. Are they bound to the module still? So I should be able to do

`(client as any).listCodes()` to call the method from the `wasmd` module, or any. Maybe it would be easier to subspace or prefix those to make it more clear. We do use `authAccounts` here not `accounts` to clarify which module is being called.

Reply:
The examples in tests clarified this for me. THanks

Reply:
With the latest update, the modules are prefixed (`client.wasm.getContractInfo`, `client.auth.account` etc.)